packages/@storefront/core/test/unit/services/url.ts

```diff
@@ -45,541 +63,266 @@ suite('URL Service', ({ expect, spy, stub, itShouldBeCore, itShouldExtendBaseSer
       expect(beautifierFactory).to.be.calledWith(app, routesWithBase);
       expect(service.beautifier).to.eq(userBeautifier);
     });
-
-    it('should add popstate listener', () => {
-      expect(addEventListener).to.be.calledWith('popstate', service.rewind);
-    });
   });
 
   describe('init()', () => {
-    it('should listenForHistoryChange and handleUrl', () => {
-      const listenForHistoryChange = service.listenForHistoryChange = spy();
-      const handleUrl = service.handleUrl = spy();
+    it('should call initHistory in flux', () => {
+      const initHistory = spy();
+      const initialUrl = 'www.example.com/ayylmao';
+      const pushState = spy(function() { expect(this).to.eq(service) });
+      const replaceState = spy(function() { expect(this).to.eq(service) });
+      const build = spy(function() { expect(this).to.eq(service) });
+      const parse = spy(function() { expect(this).to.eq(service) });
+      service.build = build;
+      service.parse = parse;
+      stub(service, 'initialUrl').returns(initialUrl);
+      stub(service, 'pushState').returns(pushState);
+      stub(service, 'replaceState').returns(replaceState);
+      service['app'].flux = <any>{ initHistory };
 
       service.init();
-
-      expect(listenForHistoryChange).to.be.calledOnce;
-      expect(handleUrl).to.be.calledOnce;
-    });
-  });
-
-  describe('handleUrl()', () => {
-    it('should call handleCurrentLocation', () => {
-      const handleCurrentLocation = service.handleCurrentLocation = spy();
-
-      service.handleUrl();
-
-      expect(handleCurrentLocation).to.be.calledOnce;
-    });
-
-    it('should warn on error', () => {
-      const href = 'https://example.com/route';
-      const route = '/route';
-      const request = 'request';
-      const parse = stub().throws();
-      const warn = spy();
-      win.location = { href };
-      service.beautifier = <any>{ parse };
-      service['app'] = <any>{ log: { warn } };
-
-      service.handleUrl();
-
-      expect(parse).to.be.calledWith(href);
-      expect(warn).to.be.called;
+      const initHistoryArgs = initHistory.getCall(0).args[0];
+
+      expect(initHistory).to.be.called;
```

Comment:
You should expect it to be called first before extracting the args of the first call.